backend/iterative_guide_improver.py

```
@@ -0,0 +1,516 @@
+    
+    def __init__(self):
+        self.guide_service = OptimizedGuideService()
+        self.pdf_renderer = HTMLPDFRenderer()
+        self.scorer = ComprehensiveGuideScorer()
+        self.max_iterations = 5
+        self.target_score = 85.0
+        
+    async def improve_guide_iteratively(
+        self,
+        trip_data: Dict[str, Any],
+        output_dir: Path = None
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)